Update index.md (#149)
esi/esi-docs: docs/services/esi/index.md

## User-Agent

> All ESI Requests **should** be accompanied by an appropriate User-Agent information providing a method for CCP to contact the source of the request in the event of an issue.

Reviewer: ```suggestion
All ESI Requests **should** be accompanied by appropriate User-Agent information providing a method for CCP to contact the source of the request in the event of an issue.
```

"an" no longer makes grammatical sense as "information" is an uncountable noun

### Data transmission

That data should be transmitted in, by order of preference 
 1. A `User-Agent` header in the request
 2. A `X-User-Agent` header instead, if that previous one can't be set reliably.

>  3. A `user_agent` query parameter, if you can't set your headers reliably.

Reviewer: Would be more consistent to keep the same voice here, either "if you can't sent the previous one reliably" in point 2, or "if headers can't be set reliably" in point 3.

Author: yeah it's spec, should be neutral.

> That data should be transmitted in, by order of preference 
>  1. A `User-Agent` header in the request
>  2. A `X-User-Agent` header instead, if that previous one can't be set reliably.
>  3. A `user_agent` query parameter, if you can't set your headers reliably.

Reviewer 2: ```suggestion
The User-Agent should be sent via one of the following methods:
- For browser applications: use `X-User-Agent` header (see "Note" section on [MDN}(https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_request_header) for the reason why).
- For applications that can't set headers: use `user_agent` query parameter (`?user_agent=...`).
- In all other cases: use `User-Agent` header.
```

I would suggest something like this. Makes it more clear when to use which. Other suggestions/ideas are welcome, ofc.

Reviewer: It might also not be obvious to everyone that in case of the query parameter, it needs to be URL encoded.

Author: Nah, I prefer to have a spec-first approach, with most common cases first, and explanation on optional use cases limited. Because the use cases are only limited by imagination, meaning if we start iterating over all use cases it would add a lot of noise for people trying to implement the most-recommended protocol.

Maybe a sub part "When you can't set your User-Agent"  ? But IMO it's important to have User-Agent header first.

What's more, the information provided are not necessarily user-agent, typically when set in the query it's a parameter. Hence the wording "that data" (which could be replace by "this information" ). The important thing is to transmit that information, the user-agent is a way (and the recommended one) but not the goal in itself.

Also if it's a url parameter, it must be url encoded ? I mean it's implied by "url parameter" ?

Reviewer: > Also if it's a url parameter, it must be url encoded ? I mean it's implied by "url parameter" ?

Many third party devs are beginners who might just construct the URL with string concatenation in their 20 line Python script without knowing it's not enough.

Author: > Many third party devs are beginners who might just construct the URL with string concatenation in their 20 line Python script without knowing it's not enough. (Granted, in a Python script they shouldn't need to use the query parameter, but to illustrate the point)

yeah, I added a link 
 3. A `user_agent` query parameter, if the previous headers can't be set reliably. Don't forget to [url-encode](https://www.rfc-editor.org/rfc/rfc3986#section-2.1) the value.

### Informations to transmit

This data transmitted should contain one or more of the following :

- An Email Address (Strongly Preferred)
- An App Name (Strongly Preferred)
- A URL to Source Code
- A Discord UID
- An EVE Character

### Recommended format

While User Agents are not a defined web standard, the MDN provides a thoroughly documented set of examples <https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent>, Following these standards will ensure your useragent is well understood.

#### Basic format

```text
AppName/1.2.3 (foo@example.com)
```

> #### Additional informations

Reviewer: ```suggestion
#### Additional information
```

The plural of "information" is "information"

You may, if you choose, include any additional information to identify your app. Following the MDN standard.

- A URL to your documentation or source code, remembering to prefix `+` on a URL `(+https://github.com/)`
- A Discord Username `(discord:username)`
- An EVE Character `(eve:charactername)`

```text

> AppName/1.2.3 (foo@example.com; +https://github.com/ eve:the L33T guy)

Reviewer: ```suggestion
AppName/1.2.3 (foo@example.com; +https://github.com/; eve:the L33T guy)
```
